types/pub_rand_commit.go

```diff
@@ -1,11 +1,34 @@
 package types
 
 import (
+	"fmt"
+
 	bbn "github.com/babylonlabs-io/babylon/types"
 	"github.com/btcsuite/btcd/btcec/v2"
 	"github.com/cometbft/cometbft/crypto/merkle"
 )
 
+type PubRandCommit struct {
+	StartHeight uint64 `json:"start_height"`
+	NumPubRand  uint64 `json:"num_pub_rand"`
+	Commitment  []byte `json:"commitment"`
+}
+
+// Validate checks if the PubRandCommit structure is valid
+// returns an error if not.
+func (prc *PubRandCommit) Validate() error {
+	if prc.NumPubRand < 1 {
+		return fmt.Errorf("NumPubRand must be >= 1, got %d", prc.NumPubRand)
+	}
+
+	return nil
+}
+
+// EndHeight returns the last height covered by this public randomness commitment
+func (prc *PubRandCommit) EndHeight() uint64 {
+	return prc.StartHeight + prc.NumPubRand - 1
```

Comment:
what if `prc.StartHeight == 0` and  `prc.NumPubRand == 0`?

Reply:
`NumPubRand` should be at least 1 and `Validate()` checks this [ref](https://github.com/babylonlabs-io/finality-provider/blob/c6f6fc1c88287d3eb995f2ce561f8069e8b04ffa/types/pub_rand_commit.go#L17-L26). The check is performed in every controller ([Babylon](https://github.com/babylonlabs-io/finality-provider/blob/c6f6fc1c88287d3eb995f2ce561f8069e8b04ffa/clientcontroller/babylon/consumer.go#L349-L350), [Cosmos](https://github.com/babylonlabs-io/finality-provider/blob/c6f6fc1c88287d3eb995f2ce561f8069e8b04ffa/clientcontroller/cosmwasm/consumer.go#L323-L324), [OP stack](https://github.com/babylonlabs-io/finality-provider/blob/c6f6fc1c88287d3eb995f2ce561f8069e8b04ffa/clientcontroller/opstackl2/consumer.go#L508-L509)).